src/debug/ee/arm/dbghelpers.S

```diff
@@ -0,0 +1,61 @@
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information. 
+//
+
+#include "unixasmmacros.inc"
+
+//
+// hijacking stub used to perform a func-eval, see Debugger::FuncEvalSetup() for use.
+//
+// on entry:
+//   r0  : pointer to DebuggerEval object
+//
+
+NESTED_ENTRY FuncEvalHijack, _TEXT, FuncEvalHijackPersonalityRoutine
+
+    // NOTE: FuncEvalHijackPersonalityRoutine is dependent on the stack layout so if
+    //       you change the prolog you will also need to update the personality routine.
+
+    // push arg to the stack so our personality routine can find it
+    // push lr to get good stacktrace in debugger
+    push {r0,lr}
+
+    //CHECK_STACK_ALIGNMENT
+
+    // FuncEvalHijackWorker returns the address we should jump to. 
+    bl      FuncEvalHijackWorker
+
+    // effective NOP to terminate unwind
+    mov r2, r2
+
+    add sp, sp, #8
+    bx r0
+
+NESTED_END FuncEvalHijack, _TEXT
+
+//
+// This is the general purpose hijacking stub. DacDbiInterfaceImpl::Hijack() will
+// set the registers with the appropriate parameters from out-of-process.
+//
+// on entry:
+//   r0 : pointer to CONTEXT
+//   r1 : pointer to EXCEPTION_RECORD
+//   r2 : EHijackReason
+//   r3 : void* pdata
+//
+
+NESTED_ENTRY ExceptionHijack, _TEXT, ExceptionHijackPersonalityRoutine
+
+    //CHECK_STACK_ALIGNMENT
+
+    // make the call
+    bl ExceptionHijackWorker
+
+    // effective NOP to terminate unwind
+    mov r3, r3
+
+    // *** should never get here ***
+    bkpt
+
+NESTED_END ExceptionHijackEnd, _TEXT    
```

Comment:
Yeah I started using the macros to allocate/free space on the stack, but I got a bit stumped when it came to push / pop since we do this with a list of registers, don't know if there is a way to count the registers.